### PR TITLE
Fixing skipped unit tests - random, solving issue #177

### PR DESCRIPTION
**Pull request by Zbysekz**
Sample and shuffle tests fixed by changing runtime_error by invalid_argument in c++ code.
Random pickling - i am not sure what more else to test here, i can expand when i will be able to run cov-test
solving issue #177

### CODE REVIEW

**Comment by breznak** [comment]
The random picke is failing on OSX, Linux CI 
https://travis-ci.org/htm-community/nupic.cpp/builds/544681573#L2642 
please have a look

**Comment by Zbysekz** [comment]
hmm yes, it seems that segmentation fault occurs only for python2 (i am using py3.6) AppVeyor also succeeded. I will take a look why is this happening

File: bindings/py/tests/nupic_random_test.py [hunk 1]
```diff
@@ -201,34 +202,41 @@ def testShuffleEmpty(self):
...
+    
     r2 = Random(42)
     v2 = r2.getReal64()
     i2 = r2.getUInt32()
+    s2 = r2.getSeed()
+    m2 = r2.min()
+    n2 = r2.max()
+    
     self.assertEqual(v1, v2)
-    self.assertEqual(r1, r2)
```
Comment (breznak) [inline bindings/py/tests/nupic_random_test.py]: why is the (Random) r comparison replaced by s,m,n? (and is it adequate?)

Comment (Zbysekz) [reply bindings/py/tests/nupic_random_test.py]: We cant expect in python that two instances of Random() will be equal, like r1=r2. 
In C++ code of Random class there is operator==() but i dont know how to bind this to python. So i just added more inner variable comparism..  Any thoughts how to solve in better way?

Comment (breznak) [reply bindings/py/tests/nupic_random_test.py]: > cant expect in python that two instances of Random() will be equal, like r1=r2.
In C++ code of Random class there is operator==()  

as you're saying, in c++ the operator works well, so we should provide it to python. 

https://developer.lsst.io/v/DM-9089/coding/python_wrappers_for_cpp_with_pybind11.html#wrapping-operators

I think `def __eq__(const Random& self, const Random& other)` to something like `return self==other; `

**Comment by ctrl-z-9000-times** [comment]
FYI: Python pickling is a known issue.  It does not work for many of the C++ classes with python bindings.  I think for this PR you can skip the pickle tests.

Thank you for working on this!

File: bindings/py/tests/nupic_random_test.py (hunk 1 above)
Comment (Zbysekz) [reply bindings/py/tests/nupic_random_test.py]: nice, thx

File: bindings/py/cpp_src/bindings/math/py_Random.cpp [hunk 2]
```diff
@@ -51,7 +51,10 @@ namespace nupic_ext {
             .def("getReal64", &Random_t::getReal64)
 			.def("getSeed", &Random_t::getSeed)
             .def("max", &Random_t::max)
-            .def("min", &Random_t::min);
+            .def("min", &Random_t::min)
+        	.def("__eq__", [](Random_t const & self, Random_t const & other) {//wrapping operator==
```
Comment (breznak) [inline bindings/py/cpp_src/bindings/math/py_Random.cpp]: nit, formatting, but we can leave that for now

**Pull request merged**